src/storage/txn/actions/cleanup.rs

```diff
@@ -33,11 +33,21 @@ pub fn cleanup<S: Snapshot>(
                 );
             }
 
+            let overlapped_write = reader.get_txn_commit_record(&key)?.unwrap_none();
             let is_pessimistic_txn = !lock.for_update_ts.is_zero();
-            txn.check_write_and_rollback_lock(key, lock, is_pessimistic_txn)
+            rollback_lock(
+                txn,
+                reader,
+                key,
+                lock,
+                is_pessimistic_txn,
+                !protect_rollback,
+                overlapped_write,
+            )
```

Comment:
I'm not very sure about this.

The origin behaviour here by `check_write_and_rollback_lock` is:
```rust
TxnCommitRecord::None => rollback_lock
TxnCommitRecord::SingleRecord => if write is not WriteType::Rollback panic!
			         else unlock_key
TxnCommitRecord::OverlappedRollback => unlock_key
```

But the new behaviour here is:
```rust
TxnCommitRecord::None => rollback_lock
_ => panic!
```

Did I miss anything?

Reply:
I think the change is from my latest PR https://github.com/tikv/tikv/pull/9656, and this PR is not updated yet.